harmonica/tests/test_prism.py

```diff
@@ -0,0 +1,356 @@
+        "C": ([-50, 50, 0, 0], [0, 0, -50, 50], [50, 50, 50, 50]),
+        "D": ([-50, 50, 0, 0], [0, 0, -50, 50], [-50, -50, -50, -50]),
+        "E": ([-50, -50, 50, 50], [-50, 50, -50, 50], [0, 0, 0, 0]),
+        "F": ([-50, -50, 50, 50], [-50, 50, -50, 50], [50, 50, 50, 50]),
+        "G": ([-50, -50, 50, 50], [-50, 50, -50, 50], [-50, -50, -50, -50]),
+    }
+    # Compute g_z on each set of points
+    results = {}
+    for group in computation_points:
+        results[group] = prism_gravity(
+            computation_points[group], prism, density, field="g_z"
+        )
+    # Check symmetries
+    # Values on A must be opposite, and the value of g_z at the point above the center
+    # of the prism must have the same sign as the density, while the one bellow should
+    # have the opposite
+    npt.assert_allclose(results["A"][0], -results["A"][1])
+    npt.assert_allclose(np.sign(results["A"][0]), -np.sign(density))
+    npt.assert_allclose(np.sign(results["A"][1]), np.sign(density))
+    # Values on C, D, F, G must be all equal within each set
+    for group in "C D F G".split():
+        npt.assert_allclose(results[group][0], results[group])
+    # Values on B and E must be zero
+    for group in "B E".split():
+        npt.assert_allclose(0, results[group])
+    # Values on C and D, F and G must be opposite
+    # Moreover, the set of points that are above the center of the prism must have the
+    # same sign as the density, while the ones bellow should have the opposite
+    for above, bellow in (("C", "D"), ("F", "G")):
+        npt.assert_allclose(results[above], -results[bellow])
+        npt.assert_allclose(np.sign(results[above]), np.sign(density))
+        npt.assert_allclose(np.sign(results[bellow]), -np.sign(density))
+
+
+@pytest.mark.use_numba
+def test_safe_atan2():
+    "Test the safe_atan2 function"
+    # Test safe_atan2 for one point per quadrant
+    # First quadrant
+    x, y = 1, 1
+    npt.assert_allclose(safe_atan2(y, x), np.pi / 4)
+    # Second quadrant
+    x, y = -1, 1
+    npt.assert_allclose(safe_atan2(y, x), -np.pi / 4)
+    # Third quadrant
+    x, y = -1, -1
+    npt.assert_allclose(safe_atan2(y, x), np.pi / 4)
+    # Forth quadrant
+    x, y = 1, -1
+    npt.assert_allclose(safe_atan2(y, x), -np.pi / 4)
+    # Test safe_atan2 if the denominator is equal to zero
+    npt.assert_allclose(safe_atan2(1, 0), np.pi / 2)
+    npt.assert_allclose(safe_atan2(-1, 0), -np.pi / 2)
+    # Test safe_atan2 if both numerator and denominator are equal to zero
+    assert safe_atan2(0, 0) == 0
+
+
+@pytest.mark.use_numba
+def test_safe_log():
+    "Test the safe_log function"
+    # Check if safe_log function satisfies safe_log(0) == 0
+    assert safe_log(0) == 0
+    # Check if safe_log behaves like the natural logarithm in case that x != 0
+    x = np.linspace(1, 100, 101)
+    for x_i in x:
+        npt.assert_allclose(safe_log(x_i), np.log(x_i))
+
+
+def gravity_infinite_slab(thickness, density):
```

Comment:
Why not use `harmonica.bouguer_correction`?

Reply:
Because it's funnier to rewrite code haha. I forgot it existed, thanks for pointing it out.

Reply:
Done!